tasks/f3/f3_task.go

```diff
@@ -136,101 +128,97 @@ func (f *F3Task) tryGetF3ParticipationTicket(ctx context.Context, stillOwned fun
+			return nil // Return to get a new ticket
 		case errors.Is(err, api.ErrF3ParticipationTicketInvalid):
 			log.Errorw("F3 participation ticket is not valid. Acquiring a new ticket after backoff.", "err", err)
 			time.Sleep(1 * time.Second)
-			return api.F3ParticipationLease{}, false, nil
+			return nil // Return to get a new ticket
 		case errors.Is(err, api.ErrF3ParticipationIssuerMismatch):
-			log.Warnw("Node is not the issuer of F3 participation ticket. Miner maybe load-balancing or node has changed. Retrying F3 participation after backoff.", "err", err)
+			log.Warnw("Node is not the issuer of F3 participation ticket. Miner may be load-balancing or node has changed. Retrying F3 participation.", "err", err)
 			time.Sleep(1 * time.Second)
 			continue
 		case errors.Is(err, api.ErrF3NotReady):
-			log.Warnw("F3 is not ready. Retrying F3 participation after backoff.", "err", err)
+			log.Warnw("F3 is not ready. Retrying F3 participation.", "err", err)
 			time.Sleep(30 * time.Second)
 			continue
 		case err != nil:
 			log.Errorw("Unexpected error while attempting F3 participation. Retrying after backoff", "err", err)
-			return api.F3ParticipationLease{}, false, xerrors.Errorf("attempting F3 participation with ticket: %w", err)
+			return xerrors.Errorf("attempting F3 participation with ticket: %w", err)
+		case lease.ValidityTerm <= renewLeaseWithin:
+			return nil // Return to get a new ticket
 		default:
-			log.Infow("Successfully acquired F3 participation lease.",
-				"issuer", lease.Issuer,
-				"not-before", lease.FromInstance,
-				"not-after", lease.ToInstance(),
-			)
-			return lease, true, nil
+			// Successfully participated
+			if !haveLease {
+				log.Infow("Successfully acquired F3 participation lease.",
+					"issuer", lease.Issuer,
+					"not-before", lease.FromInstance,
+					"not-after", lease.ToInstance(),
+				)
+				haveLease = true
+			}
 		}
-	}
-	return api.F3ParticipationLease{}, false, ctx.Err()
-}
 
-func (f *F3Task) awaitLeaseExpiry(ctx context.Context, stillOwned func() bool, lease api.F3ParticipationLease) error {
-	backoff := &backoff.Backoff{
-		Min:    1 * time.Second,
-		Max:    1 * time.Minute,
-		Factor: 1.5,
+		// Fetch the manifest if necessary
+		if manifest == nil || lease.Network != manifest.NetworkName {
+			manifest, err = f.getManifest(ctx)
+			if err != nil {
+				return err
+			}
+			if manifest.NetworkName != lease.Network {
+				log.Warnf("Got a manifest for network %q while expecting network %q. Getting another ticket.", manifest.NetworkName, lease.Network)
+				return nil // Return to get a new ticket
+			}
+		}
+
+		// Wait until we think we may need to renew the lease
+		waitTime := time.Duration(lease.ValidityTerm-renewLeaseWithin) * manifest.CatchUpAlignment
+		if waitTime <= 0 {
+			waitTime = 100 * time.Millisecond
+		}
+		log.Debugf("F3 participation lease is valid for further %d instances. Re-checking after %s.", lease.ValidityTerm, waitTime)
+		time.Sleep(waitTime)
```

Comment:
Note: if you don't want to bother fetching the manifest, you can also just wait a second (or 5) here between polls. We're doing it this way in lotus because waiting a fixed amount of time makes testing difficult.

Reply:
Ah nice, simple good